tests/phpunit/includes/testcase-trait.php

```diff
@@ -63,6 +63,20 @@ public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) { // p
 
 		remove_action( 'current_screen', '_load_remote_block_patterns' );
 		remove_action( 'current_screen', '_load_remote_featured_patterns' );
+
+		/**
+		 * Don't trigger an error if `PLL_Model::get_languages_list()` is called too early.
+		 *
+		 * @see PLL_UnitTestCase_Trait::doing_it_wrong_run()
+		 */
+		add_filter(
+			'doing_it_wrong_trigger_error',
+			function ( $trigger_error, $function_name ) {
+				return $trigger_error && 'PLL_Model::get_languages_list()' !== $function_name;
+			},
+			10,
+			2
+		);
```

Comment:
I did not test but [this](https://github.com/WordPress/wordpress-develop/blob/6.1.1/tests/phpunit/includes/abstract-testcase.php#L557) makes me think that this is useless. If I'm wrong, then I would prefer not to use an anonymous function.

Reply:
It is required because `expectDeprecated()` is called in `set_up()`, which is too late: we need this filter in `wpSetUpBeforeClass()`, before creating the languages, because `PLL_UnitTestCase::create_language` calls `PLL_Admin_Model::add_language()`, which calls `PLL_Admin_Model::validate_lang()`, then `PLL_Model::get_languages_list()`.

Reply:
Simplified and added a comment with [8e4000c](https://github.com/polylang/polylang/pull/1211/commits/8e4000c2f3317386ac67244e10d4a192dcf05643).